### PR TITLE
allow right controls in filter bar

### PR DESCRIPTION
**Pull request by evinism**
### Summary
Adds in a possible right control to the filter bar in list-layout. In my screenshot, toggler between the a thumbnail and list view.

#### Before
screen shot 2017-09-22 at 9 29 40 am

#### After
screen shot 2017-09-22 at 9 24 42 am

### Review Notes/ Questions/ TODOs
The toggler itself should probs be a particle, i'll do that too.

### CODE REVIEW

File: list-partial/src/ListPartial.jsx [hunk 1]
```diff
@@ -24,6 +25,12 @@ const ListPartial = ({
     </div>
   );
 
+  const rightControl = rightControlComponent && (
+    <div>
```
Comment (ben-burlingham) [inline list-partial/src/ListPartial.jsx]: nb Perhaps consider putting a class on this wrapper so devs can target and override the `slotRight`?

**Comment by ben-burlingham** [comment]
Also, have you considered placing this in the filter component instead?  The current thinking of the list partial is to sort of "stack together" particles for a layout, rather than customize the layout. What do you guys think with respect to your tool?

**Comment by evinism** [comment]
@nickwmurphy can do, thanks!

@ben-burlingham i chose this path because
1: it's not a breaking change
2: given how search and filter are semantically separated from each other, it seemed silly to stick what in my use case is a view switch into the same slot as a filter
3: the same thing is done for the left slot in this component

that relatively sensible, or would you still prefer preventing a code change by sticking it in the filter param over this?

**Comment by ben-burlingham** [comment]
Nope, makes sense! Thanks!

**Comment by evinism** [comment]
whoops, forgot to rebuild index, don't merge until i've done that.

**Pull request merged**